src/mqt/bench/qiskit_helper.py

```diff
@@ -78,16 +102,44 @@ def get_indep_level(
     )
 
 
+@overload
+def get_native_gates_level(
+    qc: QuantumCircuit,
+    gate_set_name: str,
+    num_qubits: int | None,
+    opt_level: int,
+    file_precheck: bool,
+    return_qc: Literal[True] = True,
+    target_directory: str = "./",
+    target_filename: str = "",
+) -> bool:
```

Comment:
```suggestion
) -> QuantumCircuit:
```
this is mixed up. The variant here should return a quantum circuit, the other should return bool